tools/orca/src/model.rs

```diff
@@ -1,15 +1,37 @@
 use crate::error::Error;
 use crate::run::{EventDetail, EventRecord};
 use crate::state::*;
+use std::str::FromStr;
 use std::time::{Duration, Instant};
 
-use kanidm_client::KanidmClient;
+use kanidm_client::{ClientError, KanidmClient};
 
 use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+// use strum_macros::EnumCount;
 
+// #[derive(EnumCount)]
 pub enum TransitionAction {
-    Login,
-    Logout,
+    Login = 0,
+    Logout = 1,
+    ReadAttribute = 2,
+    WriteAttribute = 3,
+}
+
+impl TryFrom<i32> for TransitionAction {
```

Comment:
Why does this need i32?

Reply:
Well ... it doesn't. It was the result of some VERY poor design choices that I've since abandoned :p